src/sas/qtgui/Perspectives/Inversion/InversionPerspective.py

```diff
@@ -111,21 +85,83 @@ def __init__(self, parent=None, data=None):
         self.isBatch = False
         self.batchResultsWindow = None
         self.batchResults = {}
-        self.batchComplete = []
-
-        # Add validators
-        self.setupValidators()
-        # Link user interactions with methods
-        self.setupLinks()
-        # Set values
-        self.setupModel()
-        # Set up the Widget Map
-        self.setupMapper()
-
-        #Hidding calculate all buton
-        self.calculateAllButton.setVisible(False)
-        # Set base window state
-        self.setupWindow()
+
+        # Max index for adding new, non-clashing tab names
+        self.maxIndex = 1
+
+        # The tabs need to be closeable
+        self.setTabsClosable(True)
+
+        # The tabs need to be movable
+        self.setMovable(True)
+
+        # Initialize the first tab
+        self.addData(None)
+
+    ######################################################################
+    # Batch Mode and Tab Functions
+
+    def calculate(self):
+        # Default to background estimate
+        self._calculator.est_bck = True
+
+    def resetTab(self, index):
+        """
+        Adds a new tab and removes the last tab
+        as a way of resetting the fit tabs
+        """
+        # If data on tab empty - do nothing
+        if index in self.tabs and not self.tabs[index].data:
+            return
+        # Remove the previous last tab
+        self.tabCloses(index)
+
+    def tabCloses(self, index):
+        """
+        Update local bookkeeping on tab close
+        """
+        # don't remove the last tab
+        if len(self.tabs) <= 1:
+            return
+        self.closeTabByIndex(index)
+
+    def closeTabByIndex(self, index):
+        """
+        Close/delete a tab with the given index.
+        No checks on validity of the index.
+        """
+        try:
+            self.removeTab(index)
+            del self.tabs[index]
+            self.tabsModifiedSignal.emit()
+        except IndexError:
+            # The tab might have already been deleted previously
+            pass
+
+    ######################################################################
+
+    def serializeAll(self):
```

Comment:
This will only save the active inversion tab into project files. A list of active indices should be stored in the perspective, iterating through the indices here, serializing each tab, and then combining the serialized data into a single dictionary.

According to `GuiManager.py` line 761:
`# Analysis should return {data-id: serialized-state}`

Reply:
I haven't worked with any of the serialising, exporting and importing sides of things so this slipped under the radar. IDing things is something that I will  needed to add as tabs or data currently is not IDed, Working on getting this fixed.

Reply:
If you assign a unique id to each tab (similar to M1, M2, etc for the fitting perspective), you can capture that for each serialized page in `serializeCurrentPage`.
```
def serializeAll(self):
    state = {}
    for index in self.tab_ids:  # tab_ids would need to be populated somewhere
        state.update(self.serializeCurrentPage(index))
    return state
```

Reply:
Right now it gives the file name as tab_name. Is there any convention on it?